modules/core/perf/perf_mat.cpp

```diff
@@ -96,3 +96,31 @@ PERF_TEST_P(Size_MatType, Mat_Clone_Roi,
 
     SANITY_CHECK(destination, 1);
 }
+
+///////////// Transform ////////////////////////
+
+PERF_TEST_P(Size_MatType, Mat_Transform,
+            testing::Combine(testing::Values(TYPICAL_MAT_SIZES),
+                             testing::Values(CV_8UC3, CV_8SC3, CV_16UC3, CV_16SC3, CV_32SC3, CV_32FC3, CV_64FC3))
+            )
+{
+    const Size_MatType_t params = GetParam();
+    const Size srcSize0 = get<0>(params);
+    const Size srcSize = Size(1, srcSize0.width*srcSize0.height);
+    const int type = get<1>(params);
+
+    Mat src(srcSize, type), mtx(Size(4, 3), CV_32FC1), dst(srcSize, type), mask = Mat::ones(srcSize, type) * 0xffffff;
+    declare.in(src, WARMUP_RNG).out(dst);
+    randu(mtx, 0, 1);
+    if (CV_MAT_DEPTH(type) < CV_32F)
+    {
+        src &= mask;
```

Comment:
Why do we need this?

Reply:
On some situation, the sanity check was raising error.
It seems that the final result was larger than the maximum of ```int```
The RNG was filling all 32bits of the input, so in some cases, the transform ends in following process, 
````0x8fffffff * 1 + 0x8fffffff * 1 + 1````
which is clearly larger than the maximum of int.

If there's a way to restrict the range of the input matrix, that would be great, but this was the best way I know.

Does this make sense ?

Reply:
Thanks!

```
randu(src, 0, 0xffffff); // need to adjust max value for different types 8U/8S, 16U/16S, 32S
                         // perhaps 30 would work for all types (30*1+30*1+30*1+1)
declare.in(src).out(dst); // without WARMUP_RNG (default is WARMUP_READ)
```

Perhaps we should use fixed values of transform matrix instead of random.